oauth_test.go

```diff
@@ -0,0 +1,67 @@
+package twilio
+
+import (
+	"context"
+	"github.com/twilio/twilio-go/client"
+	"testing"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockOAuth struct{}
+
+func (m *MockOAuth) GetAccessToken(ctx context.Context) (string, error) {
+	return "mock_token", nil
+}
+
+type MockTokenAuth struct {
+	token string
+	OAuth client.OAuth
+}
+
+func (m *MockTokenAuth) FetchToken() (string, error) {
+	t := &TokenAuth{token: m.token, OAuth: m.OAuth}
+	return t.token, nil
+}
+
+func (m *MockTokenAuth) TokenExpired() bool {
+	return false
+}
+
+func TestTokenAuth_FetchToken(t *testing.T) {
+	oauth := &MockOAuth{}
+	tokenAuth := &TokenAuth{OAuth: oauth}
+
+	// Test fetching a new token
+	token, err := tokenAuth.FetchToken()
+	assert.NoError(t, err)
+	assert.Equal(t, "mock_token", token)
+
+	// Test fetching the cached token
+	mockTokenAuth := MockTokenAuth{OAuth: oauth, token: "cached_token"}
+	token, err = mockTokenAuth.FetchToken()
+	assert.NoError(t, err)
+	assert.Equal(t, "cached_token", token)
+}
+
+func TestTokenAuth_TokenExpired(t *testing.T) {
+	// Test with an expired token
+	expiredToken := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"exp": time.Now().Add(-time.Hour).Unix(),
+	})
+	expiredTokenString, _ := expiredToken.SignedString([]byte("secret"))
```

Comment:
looks like there are a few legitimate types of errors that can be generated by `SignedString` - given this is security code, I think we should assert the error is nil instead of eliding it. 

https://github.com/golang-jwt/jwt/blob/v3.2.2/token.go#L49-L83